pyvista/plotting/render_window_interactor.py

```diff
@@ -102,6 +103,8 @@ def add_key_event(self, key, callback):
         """
         if not callable(callback):
             raise TypeError('callback must be callable.')
+        if len(signature(callback).parameters) > 0:
```

Comment:
Isn't this a bit too restrictive? If the callback provides default values for all arguments, it shouldn't be a problem right? (But I'm not sure how you would check that)

Reply:
Thank you for your comment. That is a very good point. I had completely missed it's use case. The following code will accomplish that.
```suggestion
        for param in signature(callback).parameters.values():
            if (param.default is param.empty):
```